go-controller/pkg/ovn/endpoints.go

```diff
@@ -0,0 +1,124 @@
+package ovn
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"os/exec"
+	"strings"
+	"unicode"
+
+	kapi "k8s.io/client-go/pkg/api/v1"
+)
+
+func (ovn *OvnController) getLoadBalancer(protocol kapi.Protocol) string {
```

Comment:
Since the load-balancer is created before watcher gets any events and it remains constant, a cache here will save ovn-nbctl calls.